File: algorithms/php/167.TwoSumII-InputArrayIsSorted.php
<?php

class Solution
{

    /**
     * @param  Integer[]  $numbers
     * @param  Integer  $target
     * @return Integer[]
     */
    function twoSum($numbers, $target)
    {
        $l = 0;
        $r = count($numbers) - 1;

        while ($numbers[$l] + $numbers[$r] !== $target) {
            if ($numbers[$l] + $numbers[$r] < $target) {
                $l++;
            } else {
                $r--;
            }
        }

        return [$l + 1, $r + 1];
    }
}

File: algorithms/cpp/167.TwoSumII-InputArrayIsSorted.cpp
class Solution {
public:
    int binSearch(vector<int>& arr, int target, int l, int r) {
        while (l <= r) {
            int m = l + (r - l) / 2;

            if (arr[m] == target) {
                return m;
            } else if (arr[m] < target) {
                l = m + 1;
            } else {
                r = m - 1;
            }
        }

        return -1;
    }

    vector<int> twoSum(vector<int>& numbers, int target) {
        int n = numbers.size();
        for (int i = 0; i < n; i++) {
            int index = binSearch(numbers, target - numbers[i], i + 1, n - 1);
            if (index != -1) {
                return {i + 1, index + 1};
            }
        }

        return {};
    }
};